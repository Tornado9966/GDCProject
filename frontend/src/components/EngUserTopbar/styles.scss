@import 'assets/_colors.scss';

.ui {
  &.modal-button {
    background-color: $yellow;
    font-size: 16px;
    color: #fff;
    border: 1px solid $yellow;

    &:hover {
      background-color: $yellow;
      color: $dark_grey;
      border: 1px solid $dark_grey;
    }
    &:active {
      background-color: $dark_yellow;
      color: #fff;
      border: 1px solid $dark_yellow;
    }
  }
}