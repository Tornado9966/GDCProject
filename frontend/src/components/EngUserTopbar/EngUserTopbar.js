import React from 'react';
import PropTypes from 'prop-types';
import { Button, Modal } from 'semantic-ui-react';

import EngLoginForm from 'components/EngLoginForm';
import { loginBtnText }  from 'constants/text-constants';
import EngUserDropdown from 'components/EngUserDropdown';

import './styles.scss';

export function EngUserTopbar({ loggedIn }) {

  return loggedIn ?
    <EngUserDropdown /> :
    <Modal trigger={<Button className='modal-button'>{loginBtnText}</Button>}>
      <EngLoginForm />
    </Modal>;
}

EngUserTopbar.propTypes = {
  loggedIn: PropTypes.bool
};

EngUserTopbar.defaultProps = {
  loggedIn: false
};