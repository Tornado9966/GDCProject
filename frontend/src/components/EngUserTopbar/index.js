import {connect} from 'react-redux';

import { EngUserTopbar } from './EngUserTopbar';

function mapState({auth: {loggedIn}}) {
  return {
    loggedIn
  };
}
  
export default connect(mapState)(EngUserTopbar);
