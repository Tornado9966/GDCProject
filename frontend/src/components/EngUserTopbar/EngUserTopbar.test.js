import React from 'react';
import { shallow } from 'enzyme';

import { EngUserTopbar } from './EngUserTopbar';

const setUp = (props = {}) => {
    const component = shallow(<EngUserTopbar {...props} />);
    return component;
};

describe('EngUserTopbar', () => {
    let component;
    const loggedIn = true;

    beforeEach(() => {
        component = setUp();
    });

    it('matches the snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    it('should render correctly in "debug" mode', () => {
        const component = shallow(<EngUserTopbar debug />);
        expect(component).toMatchSnapshot();
    });
    
    it('should render correctly with no props', () => {
        const component = shallow(<EngUserTopbar />);
        
        expect(component).toMatchSnapshot();
    });

    it('should render correctly with boolean prop', () => {
        const component = shallow(<EngUserTopbar loggedIn={loggedIn} />);
        expect(component).toMatchSnapshot();
    });
});