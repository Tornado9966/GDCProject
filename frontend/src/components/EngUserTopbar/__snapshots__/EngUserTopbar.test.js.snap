// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EngLoginForm matches the snapshot 1`] = `ShallowWrapper {}`;

exports[`EngUserTopbar matches the snapshot 1`] = `ShallowWrapper {}`;

exports[`EngUserTopbar should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`EngUserTopbar should render correctly with boolean prop 1`] = `ShallowWrapper {}`;

exports[`EngUserTopbar should render correctly with no props 1`] = `ShallowWrapper {}`;
